lib/src/game/game.dart

```diff
@@ -22,9 +22,21 @@ abstract class Game {
   final images = Images();
   final assets = AssetsCache();
   BuildContext buildContext;
+  bool _hasLayout = false;
+
+  /// Indicates if the this game instance had its layout layed into the GameWidget
+  /// Only this is true, the game is ready to have its size used or in the case
+  /// of a BaseGame, to receive components.
+  bool get hasLayout => _hasLayout;
 
   /// Current game viewport size, updated every resize via the [resize] method hook
-  final Vector2 size = Vector2.zero();
+  Vector2 get size {
+    assert(hasLayout,
+        '"size" is not ready yet, did you tried to access it on the Game constructor?');
+    return _size;
+  }
+
+  final Vector2 _size = Vector2.zero();
```

Comment:
it seems you have two variables to control the same thing (_size, _hasLayout)
why not start _size with null and then hasLayout := size != null?

Reply:
I would actually prefer that, but at some point we decided to not have null on that variable. I would be ok on making it billable if we all agree on that

Reply:
well now that it becomes private I think it's ok, the public size getter will never return null because of the assert

Reply:
I agree, we can easily do this now!